master branch build failure (53c5b4f) with papi

I've just tried building and it failed with a number of errors, in areas of code associated with PAPI.
The output of g++ is below.  System info immediately follows:

build system: debian 3.16.7-ckt11-1 (2015-05-24) x86_64 GNU/Linux -- running on AWS

cmake command:
cmake -DCMAKE_BUILD_TYPE=DEBUG -DHPX_WITH_COMPRESSION_SNAPPY=ON -DHPX_WITH_HWLOC=ON -DHPX_WITH_PARCELPORT_MPI=OFF -DHPX_WITH_PARCELPORT_MPI_MULTITHREADED=OFF -DHPX_WITH_PARCELPORT_TCP=ON -DHPX_WITH_GOOGLE_PERFTOOLS=ON -DHPX_WITH_PAPI=ON -DBOOST_ROOT=/usr/local/src/boost_1_61_0 -DCMAKE_INSTALL_PREFIX=~/packages/hpx ..

compiler: GCC 6.1 (output of g++ -v below)
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/local/libexec/gcc/x86_64-pc-linux-gnu/6.1.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: ../configure --enable-lto --enable-gold --disable-multilib --disable-libada --disable-liboffloadmic --enable-bootstrat --enable-compressed-debug-sections=all --program-suffix=-6 --with-arch=native --enable-stage1-languages=c,c++,lto
Thread model: posix
gcc version 6.1.0 (GCC)

Package: libpapi-dev
Source: papi
Version: 5.3.2-2


Package: cmake
Version: 3.0.2-1


Package: libsnappy-dev
Source: snappy
Version: 1.1.2-3

Package: libhwloc-dev
Source: hwloc
Version: 1.10.0-3

Package: libgoogle-perftools-dev
Source: google-perftools
Version: 2.2.1-0.2


```
Scanning dependencies of target papi_counters_component
[ 64%] Building CXX object src/components/performance_counters/papi/CMakeFiles/papi_counters_component.dir/util/papi.cpp.o
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:35:9: error: expected ‘}’ before ‘{’ token
         {"kernel", PAPI_DOM_KERNEL}
         ^
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:35:9: error: expected ‘,’ or ‘;’ before ‘{’ token
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:36:9: error: expected unqualified-id before ‘{’ token
         {"other",  PAPI_DOM_OTHER}
         ^
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:37:9: error: expected unqualified-id before ‘{’ token
         {"super",  PAPI_DOM_SUPERVISOR}
         ^
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:38:9: error: expected unqualified-id before ‘{’ token
         {"all",    PAPI_DOM_ALL}
         ^
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:39:9: error: expected unqualified-id before ‘{’ token
         {"min",    PAPI_DOM_MIN}
         ^
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:40:9: error: expected unqualified-id before ‘{’ token
         {"max",    PAPI_DOM_MAX}
         ^
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp: In function ‘int hpx::performance_counters::papi::map_domain(const string&)’:
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:46:57: error: ‘papi_domain_map’ was not declared in this scope
         std::map<std::string, int>::const_iterator it = papi_domain_map.find(s);
                                                         ^~~~~~~~~~~~~~~
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:46:57: note: suggested alternative:
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:33:38: note:   ‘hpx::performance_counters::papi::util::papi_domain_map’
     std::map<std::string, int> const papi_domain_map = {
                                      ^~~~~~~~~~~~~~~
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp: At global scope:
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:58:5: error: ‘options_description’ does not name a type
     options_description get_options_description()
     ^~~~~~~~~~~~~~~~~~~
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:87:5: error: ‘variables_map’ does not name a type
     variables_map get_options()
     ^~~~~~~~~~~~~
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:102:30: error: ‘variables_map’ was not declared in this scope
     bool need_papi_component(variables_map const& vm)
                              ^~~~~~~~~~~~~
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:102:30: note: suggested alternatives:
In file included from /usr/local/src/boost_1_61_0/boost/program_options.hpp:18:0,
                 from /usr/local/src/hpx/hpx/util/parse_command_line.hpp:13,
                 from /usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:12:
/usr/local/src/boost_1_61_0/boost/program_options/variables_map.hpp:146:38: note:   ‘boost::program_options::variables_map’
     class BOOST_PROGRAM_OPTIONS_DECL variables_map : public abstract_variables_map,
                                      ^~~~~~~~~~~~~
/usr/local/src/boost_1_61_0/boost/program_options/variables_map.hpp:146:38: note:   ‘boost::program_options::variables_map’
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:314:4: error: expected ‘}’ at end of input
 }}}}
    ^
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:314:4: error: expected ‘}’ at end of input
/usr/local/src/hpx/src/components/performance_counters/papi/util/papi.cpp:314:4: error: expected ‘}’ at end of input
src/components/performance_counters/papi/CMakeFiles/papi_counters_component.dir/build.make:54: recipe for target 'src/components/performance_counters/papi/CMakeFiles/papi_counters_component.dir/util/papi.cpp.o' failed
make[2]: *** [src/components/performance_counters/papi/CMakeFiles/papi_counters_component.dir/util/papi.cpp.o] Error 1
CMakeFiles/Makefile2:2511: recipe for target 'src/components/performance_counters/papi/CMakeFiles/papi_counters_component.dir/all' failed
make[1]: *** [src/components/performance_counters/papi/CMakeFiles/papi_counters_component.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/src/components/performance_counters/papi/papi_startup.cpp
+++ b/src/components/performance_counters/papi/papi_startup.cpp
@@ -20,8 +20,14 @@
 #include <hpx/util/thread_mapper.hpp>
 #include <hpx/components/performance_counters/papi/server/papi.hpp>
 #include <hpx/components/performance_counters/papi/util/papi.hpp>
+#include <hpx/runtime.hpp>
+#include <hpx/runtime/components/component_commandline.hpp>
 #include <hpx/runtime/components/component_factory_base.hpp>
 #include <hpx/runtime/components/component_startup_shutdown.hpp>
+#include <hpx/runtime/components/server/create_component.hpp>
+#include <hpx/runtime/startup_function.hpp>
+#include <hpx/performance_counters/counter_creators.hpp>
+#include <hpx/performance_counters/manage_counter_type.hpp>
 #include <hpx/exception.hpp>
 
 ///////////////////////////////////////////////////////////////////////////////
@@ -291,7 +297,7 @@ namespace hpx { namespace performance_counters { namespace papi
         }
     }
 
-    bool check_startup(hpx::util::function_nonser<void()>& startup_func,
+    bool check_startup(hpx::startup_function_type startup_func,
         bool& pre_startup)
     {
         // PAPI initialization

--- a/src/components/performance_counters/papi/server/papi.cpp
+++ b/src/components/performance_counters/papi/server/papi.cpp
@@ -8,6 +8,7 @@
 
 #if defined(HPX_HAVE_PAPI)
 
+#include <hpx/runtime.hpp>
 #include <hpx/runtime/components/server/component.hpp>
 #include <hpx/runtime/components/derived_component_factory.hpp>
 #include <hpx/runtime/actions/continuation.hpp>

--- a/src/components/performance_counters/papi/util/papi.cpp
+++ b/src/components/performance_counters/papi/util/papi.cpp
@@ -31,12 +31,12 @@ namespace hpx { namespace performance_counters { namespace papi { namespace util
 
     // PAPI domain description strings
     std::map<std::string, int> const papi_domain_map = {
-        {"user",   PAPI_DOM_USER}
-        {"kernel", PAPI_DOM_KERNEL}
-        {"other",  PAPI_DOM_OTHER}
-        {"super",  PAPI_DOM_SUPERVISOR}
-        {"all",    PAPI_DOM_ALL}
-        {"min",    PAPI_DOM_MIN}
+        {"user",   PAPI_DOM_USER},
+        {"kernel", PAPI_DOM_KERNEL},
+        {"other",  PAPI_DOM_OTHER},
+        {"super",  PAPI_DOM_SUPERVISOR},
+        {"all",    PAPI_DOM_ALL},
+        {"min",    PAPI_DOM_MIN},
         {"max",    PAPI_DOM_MAX}
     };
 

--- a/src/components/performance_counters/papi/util/papi.cpp
+++ b/src/components/performance_counters/papi/util/papi.cpp
@@ -5,13 +5,16 @@
 
 #include <hpx/config.hpp>
 
+
 #if defined(HPX_HAVE_PAPI)
 
 #include <hpx/config/asio.hpp>
 #include <hpx/exception.hpp>
+#include <hpx/runtime.hpp>
 #include <hpx/util/parse_command_line.hpp>
 #include <hpx/util/thread_mapper.hpp>
 #include <hpx/components/performance_counters/papi/util/papi.hpp>
+
 
 #include <boost/format.hpp>
 #include <boost/asio/ip/host_name.hpp>

